packages/salesforcedx-vscode-core/src/context/workspaceContext.ts

```diff
@@ -20,16 +22,21 @@ import { workspaceContextUtils } from '.';
 export class WorkspaceContext {
   protected static instance?: WorkspaceContext;
 
+  protected workspaceStore: vscode.Memento;
   public readonly onOrgChange: vscode.Event<OrgUserInfo>;
 
   protected constructor() {
     const workspaceContextUtil = WorkspaceContextUtil.getInstance();
     this.onOrgChange = workspaceContextUtil.onOrgChange;
     this.onOrgChange(this.handleCliConfigChange);
+    this.workspaceStore = {} as vscode.Memento;
   }
 
   public async initialize(extensionContext: vscode.ExtensionContext) {
     await WorkspaceContextUtil.getInstance().initialize(extensionContext);
+    this.workspaceStore = extensionContext.workspaceState;
+    CommandEventStream.getInstance().initialize(extensionContext);
+    CommandLog.getInstance().initialize();
```

Comment:
Registering with another service doesn't seem like a business WorkspaceContext should be in.  
Move to the initializeProject function? https://github.com/forcedotcom/salesforcedx-vscode/blob/c7aa878a847eae8909f0ec6d561762e35672d700/packages/salesforcedx-vscode-core/src/index.ts#L714